Framework/CurveFitting/inc/MantidCurveFitting/FortranVector.h

```diff
@@ -0,0 +1,167 @@
+#ifndef MANTID_CURVEFITTING_FORTRANVECTOR_H_
+#define MANTID_CURVEFITTING_FORTRANVECTOR_H_
+
+#include <utility>
+
+namespace Mantid {
+namespace CurveFitting {
+
+/** FortranVector is a wrapper template for GSLVactor and ComplexVector
+  to simplify porting fortran programs to C++.
+  This vector allows to use arbitrary index bases as they do in
+  fortran. Indexing can begin with any integer number including
+  negative.
+
+  Copyright &copy; 2016 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+  National Laboratory & European Spallation Source
+
+  This file is part of Mantid.
+
+  Mantid is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 3 of the License, or
+  (at your option) any later version.
+
+  Mantid is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+  File change history is stored at: <https://github.com/mantidproject/mantid>
+  Code Documentation is available at: <http://doxygen.mantidproject.org>
+*/
+template <class VectorClass> class FortranVector : public VectorClass {
+  /// Base for the index
+  int m_base;
+  /// Typedef the types returned by the base class's operators []. They aren't
+  /// necessarily the same as the stored type (double or complex).
+  typedef decltype(
+      std::declval<const VectorClass>().operator[](0)) ElementConstType;
+  typedef decltype(std::declval<VectorClass>().operator[](0)) ElementRefType;
+
+public:
+  /// Constructor
+  FortranVector();
+  /// Constructor
+  FortranVector(const int n);
+  /// Copy constructor
+  FortranVector(const FortranVector &V);
```

Comment:
Again, this looks like a member-wise copy which the default would give you and removing it would then allow the default move operations to be defined.